crates/primitives-traits/src/transaction/signed.rs

```diff
@@ -77,3 +82,23 @@ pub trait SignedTransaction:
     /// Fills [`TxEnv`] with an [`Address`] and transaction.
     fn fill_tx_env(&self, tx_env: &mut TxEnv, sender: Address);
 }
+
+impl<T: SignedTransaction> AlloyTransactionExt for T {
+    type Type = <T::Transaction as AlloyTransactionExt>::Type;
+
+    fn signature_hash(&self) -> B256 {
+        self.transaction().signature_hash()
+    }
```

Comment:
imo we can add `SigneableTransaction` as super trait like I did initially, since all the tx types we now import from alloy, already impl this trait @mattsse